config/application.config.php

```diff
@@ -54,6 +54,8 @@
 // Enable caching unless in dev mode or running tests:
 $useCache = APPLICATION_ENV != 'development' && !defined('VUFIND_PHPUNIT_RUNNING');
 
+defined('CACHE_ENABLED') || define('CACHE_ENABLED', $useCache);
```

Comment:
Does this line (and the corresponding one in the test bootstrap) actually do anything? I don't see where CACHE_ENABLED is used as a global constant.

Reply:
Oh ... the usage somehow got lost - I intended this global constant to be used in `VuFind\Config\Manager` instead of the corresponding class constant defined there.

Reply:
Thanks for the clarification. If you want to go forward with making that all connect together, I have a few suggestions:

1.) I'd recommend changing the variable name to CONFIG_CACHE_ENABLED, since there are other kinds of caches in VuFind, and that will help reduce confusion.

2.) Rather than defaulting the constant to the value of $useCache, maybe it would be better to reverse the order of operations, defining the constant first with a default value equal to the current value of $useCache, and then set $useCache equal to the constant. That way, the constant can be used to override the behavior of all configuration-related caches.

3.) It may be worth introducing an environment variable in httpd-vufind.conf to make it easier to override this behavior from the outside.

Reply:
Your ideas sound reasonable ... I will incorporate them asap - probably tomorrow. Furthermore I intend to do the following (among other things):

1) Let the path to the configuration (by default `config/config.php`) be a parameter to `VuFind\Config\Manager` which might then be set by a (to be created) factory and could moreover be changed in test case classes.

2) Write tests for the new provider classes in `VuFind\Config\Provider` namespace.

3) Refactor the test methods in `VuFindTest\Config\SearchSpecsReader` depending on removed protected methods (i.e. `getFromPaths`).

Reply:
Sounds good to me. Thanks for the continued work on this!